src/parse.rs

```diff
@@ -180,126 +241,204 @@ impl ClosureDecorator {
+      Stmt::Break(break_stmt) => new_node(
+        Node::BreakStmt,
+        vec![break_stmt
+          .label
+          .as_ref()
+          .map(|label| self.parse_ident(label, false))
+          .unwrap_or(undefined_expr())],
+      ),
+      Stmt::Continue(continue_stmt) => new_node(
+        Node::ContinueStmt,
+        vec![continue_stmt
+          .label
+          .as_ref()
+          .map(|label| self.parse_ident(label, false))
+          .unwrap_or(undefined_expr())],
+      ),
+      Stmt::Debugger(_debugger) => new_node(Node::DebuggerStmt, EMPTY_VEC),
       Stmt::Decl(decl) => self.parse_decl(decl),
-      Stmt::DoWhile(do_while) => self.new_node(
+      Stmt::DoWhile(do_while) => new_node(
         Node::DoStmt,
         vec![
           // block
           match do_while.body.as_ref() {
             Stmt::Block(block) => self.parse_block(&block),
-            stmt => self.new_node(Node::BlockStmt, vec![
-              Box::new(Expr::Array(ArrayLit {
+            stmt => new_node(
+              Node::BlockStmt,
+              vec![Box::new(Expr::Array(ArrayLit {
                 elems: vec![Some(ExprOrSpread {
                   expr: self.parse_stmt(stmt),
                   spread: None,
                 })],
-                span: DUMMY_SP
-              }))
-            ])
+                span: DUMMY_SP,
+              }))],
+            ),
           },
           // condition
           self.parse_expr(do_while.test.as_ref()),
         ],
       ),
-      Stmt::Empty(_empty) => self.new_node(Node::EmptyStmt, EMPTY_VEC),
-      Stmt::Expr(expr_stmt) => self.new_node(
+      Stmt::Empty(_empty) => new_node(Node::EmptyStmt, EMPTY_VEC),
+      Stmt::Expr(expr_stmt) => new_node(
         Node::ExprStmt,
         vec![
           //expr
           self.parse_expr(expr_stmt.expr.as_ref()),
         ],
       ),
-      // TODO
-      Stmt::For(for_stmt) => self.new_node(
-        Node::ForStmt,
-        vec![
-          self.parse_stmt(for_stmt.body.as_ref()),
-          for_stmt
-            .init
-            .as_ref()
-            .map(|init| match init {
-              VarDeclOrExpr::Expr(expr) => self.parse_expr(expr.as_ref()),
-              VarDeclOrExpr::VarDecl(var) => self.parse_var_decl(&var),
-            })
-            .unwrap_or(undefined_expr()),
-          for_stmt.test.as_ref().map(|test| self.parse_expr(test)).unwrap_or(undefined_expr()),
-          for_stmt.update.as_ref().map(|test| self.parse_expr(test)).unwrap_or(undefined_expr())
-        ], 
-      ),
+      Stmt::For(for_stmt) => {
+        // create a unique scope for the contents for the variables declared in the for-loop's initializer
+        self.vm.enter();
+
+        let init = for_stmt
+          .init
+          .as_ref()
+          .map(|init| match init {
+            VarDeclOrExpr::Expr(expr) => self.parse_expr(expr.as_ref()),
+            VarDeclOrExpr::VarDecl(var) => {
```

Comment:
This can also be a binding pattern, I think?

Reply:
Do you have example syntax?